src/context/auth.tsx

```diff
@@ -0,0 +1,117 @@
+import React, { useContext, useReducer, useCallback } from "react";
+import axios from "axios";
+import { getUiUrl } from "utils/getEnvironmentVariables";
+
+interface State {
+  isAuthenticated: boolean;
+}
+
+const defaultState: State = {
+  isAuthenticated: false
+};
+
+type Action = { type: "authenticate" } | { type: "deauthenticate" };
+
+type Dispatch = (action: Action) => void;
+
+export type Logout = () => void;
+
+interface DispatchContext {
+  login: (LoginParams) => void;
```

Comment:
LoginParams is a param name here, but I think you meant to use it as a type. Also does it make sense to export both the Login and Logout types?

Reply:
LoginParams is a `{}` so it would be equivalent to writing `login: ({username, password}: LoginParams) => void;`

i only export things that need to be exported--Logout type is used elsewhere but LoginParams is not